tfe/github_app_installation_helpers.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tfe
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-tfe"
+)
+
+func fetchGithubAppInstallationByNameOrGHID(ctx context.Context, tfeClient *tfe.Client, name string, installationID int) (*tfe.GHAInstallation, error) {
+	// Paginate through all GithubAppInstallation; if multiple pages
+	// of results are returned by the API, use the options variable to increment
+	// the page number until all results have been retrieved.
+	//
+	// Within the pagination loop, loop again through each result on each page.
+	// If 'name' was set, then match against the 'Name' field. If 'installation_id'
+	// was set, then match against the 'installation_id' field. If both are set,
+	// then both must match. All matches are added to the ocMatches slice.
+	//
+	// At the end of the loop, if zero or more than one matches were found, an
+	// error is returned. Otherwise, only one match was found, and that match is
+	// returned.
+	//
+	var ghainsMatches []*tfe.GHAInstallation
+	options := &tfe.GHAInstallationListOptions{}
+	for {
+		ghaInstList, err := tfeClient.GHAInstallations.List(ctx, options)
+		if err != nil {
+			return nil, fmt.Errorf("Error retrieving Github App Installations: %w", err)
+		}
+
+		for _, item := range ghaInstList.Items {
+			switch {
+			case name != "" && installationID != 0:
```

Comment:
If at least one of `installation_id` or `name` is required for the fetch, it might be useful to make an initial check to ensure that both attributes are not set to these empty values and immediately return without making a fetch attempt. Currently, this check is made only after a fetch has been done.